### ISSUE TITLE
[python-ray] use of uninitialized updater_io_service_

### ISSUE THREAD

**Issue by acxz** [acxz/pkgbuilds#191]
```
Use --sandbox_debug to see verbose messages from the sandbox
src/ray/core_worker/gcs_server_address_updater.cc: In constructor 'ray::core::GcsServerAddressUpdater::GcsServerAddressUpdater(std::string, int, std::function<void(std::__cxx11::basic_string<char>, int)>)':
src/ray/core_worker/gcs_server_address_updater.cc:24:28: error: member 'ray::core::GcsServerAddressUpdater::updater_io_service_' is used uninitialized [-Werror=uninitialized]
   24 |     : client_call_manager_(updater_io_service_),
      |                            ^~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

**Comment by acxz** [comment]
upstream issue: https://github.com/ray-project/ray/issues/24984

**Comment by acxz** [comment]
patched with 4eeefb3

**Comment by acxz** [comment]
Patch is no longer needed
